Feature/frontend dependencies updates

achen2401 opened the pull request:
https://app.shortcut.com/cirg/story/2107/femr-frontend-library-updates
general frontend library upgrades
- use Node `20` from `12` for deployment
- update React to  version`18` from `16`
- update Material UI library from version `4` to `6`
- update Testing library (using React testing library)
- update Webpack library from version`4` to `5` (latest version)  and its associated dependent libraries

Other fixes
- code linting
- minor responsive styling fixes

Tested and functioning in my own local instance

achen2401 commented:
@ivan-c not urgent, but the frontent lint test is failing, I think it is because [this file](https://github.com/uwcirg/cosri-patientsearch/blob/develop/.github/workflows/lint.yaml#L18) needs to be updated, e.g. using higher node version, but I don't have permission to modify the file.  Error I received: 
```
! [remote rejected] HEAD -> feature/frontend-dependencies-updates (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/lint.yaml` without `workflow` scope)
```
Can you grant me access when you have a moment?

ivan-c commented:
@achen2401  sorry for not responding sooner, are you still seeing that error?

>refusing to allow a Personal Access Token to create or update workflow

If I understand correctly, that error is coming from github, when you run `git push` from your own machine, ie not in the GitHub Actions CI log.

I did a quick google and I think you may need to add the `workflow` scope to the Personal Access Token you're using with `git push`.

I didn't have any PATs to test, but I think they should be listed below:

https://github.com/settings/tokens

Just in case, I also added you as an admin to uwcirg/cosri-patientsearch

achen2401 commented:
> are you still seeing that error?
> 
> > refusing to allow a Personal Access Token to create or update workflow
> 
> If I understand correctly, that error is coming from github, when you run `git push` from your own machine, ie not in the GitHub Actions CI log.
> 
> I did a quick google and I think you may need to add the `workflow` scope to the Personal Access Token you're using with `git push`.
> 
> I didn't have any PATs to test, but I think they should be listed below:
> 
> https://github.com/settings/tokens
> 
> Just in case, I also added you as an admin to uwcirg/cosri-patientsearch

Thank you so much @ivan-c for the helpful advice. I was still getting the error but no longer after I added the `workflow` scope to the Personal Access Token you're using with `git push`, as you suggested.  Frontend test is now passing.

achen2401 commented:
Noting that this has passed QA by Sie-ce.  Code changes are ready for review.

ivan-c commented on .github/workflows/lint.yaml:
```diff
@@ -10,19 +10,19 @@ jobs:
 
       - name: Run linters
-        uses: wearerequired/lint-action@v1
+        uses: rkuykendall/lint-action@master
```
I think you switched to this fork for node 20 support. looks like a good fix, but the new author is a bit of an unknown since they're an individual vs a company (wearerequired). unclear if the author will maintain this fork going forward, or what direction they'll take it in.

they didn't release any git tags, so I used a git commit hash to make sure any new versions don't break CI unexpectedly. there's also [some security reasons to use a git hash](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions), but I'm not convinced we need to make the same change everywhere yet.

let me know if I'm missing something about though

```suggestion
        # TODO switch back to wearerequired/lint-action when node 20 supported
        uses: rkuykendall/lint-action@3402def6dea8dbcf0b8ec42431d9345ee3f69c4d
```

ivan-c reviewed (approved):
Looks great, thank you!!

Had a minor nitpick about the lint-action, but not a blocker
